### PR TITLE
Cache and prefetch shield files

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract

When creating or loading a shield wallet, a bunch of helper scripts are fetched from the server sequentially. This happens one time for each thread.
Even though they're cached, a request is still sent to the server to see if the file changed, this creates a noticeable between creating a wallet and getting to the dashboard, particularly on machines with a high number of cores, and/or with a high network latency.

This PR aims to prefetch the required shield files, before they're actually needed. The idea is that by the time someone has created and backed up their seedphrase, the browser would have already fetched the files.
Additionally, this PR adds the util scripts each worker needs in the sapling cache to avoid contacting the server for each worker.

## Testing
- Have a third world connection (or simulate it using the devtools)
- Open MPW and try to create a new wallet or import an existing one, and see if there is any noticeable lag

### CODE REVIEW

File: scripts/native-worker.js [hunk 1]
```diff
@@ -13,30 +13,49 @@ self.addEventListener('activate', (_event) => {
     // Tell the active service worker to take control of the page immediately.
     self.clients.claim();
 });
+self.addEventListener('message', (event) => {
+    console.log(event.data);
```
Comment (panleone) [inline scripts/native-worker.js:17]: remove this console.log

File: scripts/native-worker.js [hunk 2]
```diff
@@ -13,30 +13,49 @@ self.addEventListener('activate', (_event) => {
...
+    if (event.data) populatePrefetchCache(event.data);
+});
+
+async function populatePrefetchCache(files) {
+    console.log(files);
```
Comment (panleone) [inline scripts/native-worker.js:22]: remove this console.log

**Review by panleone — CHANGES_REQUESTED** [review]
nice PR

**Review by panleone — APPROVED** [review]
tACK e6191beeb534c4fdb4cf9f08e431d48d5ec8d563

**Review by JSKitty — CHANGES_REQUESTED** [review]
Receiving error after simply opening the page, Firefox on Linux.

```
Uncaught (in promise) TypeError: registration.active is null
    registerWorker native.js:15
```

image

**Review by panleone — APPROVED** [review]
tACK b3c1e7e354eb2c4d39a1f449293e25c2ebe83f16

noticeably faster than the master branch

**Review by JSKitty — APPROVED** [review]
**tACK** c71fed46876d1926532af903f767db4f8d5d4656

Took a few attempts for my configuration, but undeniably faster loading speed now. Happy this was taken care of. 🙏

**Pull request merged**